Update docs readme with editing info (#9072)
supabase/supabase: apps/reference/README.md

# Reference Docs

> Home to all of the documentation for Supabase. Built with [https://docusaurus.io/](Docusaurus) and [https://typesense.org/](typesense)

Reviewer: ```suggestion
Documentation website for Supabase. Built with [https://docusaurus.io/](Docusaurus) and [https://typesense.org/](typesense)
```

The benefit of using custom specifications is that we can generate many other types from a strict schema (eg, HTML and manpages).
It also means that we can switch any documentation system we want. On this site we use Docusaurus, but in Supabase's official website we use a custom React site and expose only a subset of the available API for each tool.

## Local setup

> To run the docs locally, you need to:

Reviewer 2: External contributors will have to fork the repo too.

- Clone the main [https://github.com/supabase/supabase](Supabase repo)
  — Run `npm install` from root of the repo
- Run `npm run dev` from the root of the repo, or from the `apps/reference` directory (if you want to run the reference docs only)
- Access the docs from [http://localhost:3010/docs/](http://localhost:3010/docs/)

## How to contribute?

- Branch from `master` and name your branches with the following structure
  - `{type}/{branch_name}`
    - Type: `chore | fix | feature`
    - The branch name is arbitrary — just make sure it summarizes the work.
- When you send a PR to `master`, tag [https://github.com/dannykng](@dannykng) for a review.

> ### Which files to edit?

Reviewer 2: I don't think we're quite there yet, but the **Edit this page** button at the bottom of each doc should be the source of truth. We can circle back and add this once we have it all cleaned up.

> Our docs site are comprised of guides, tutorials, quickstarts as well as reference docs. It's important to note that these reference docs (located at https://supabase.com/docs/reference) are generated from spec files. As the DocSpec section above notes, we use documentation specs to generate human-readable docs.

Reviewer 2: ```suggestion
Our docs site are comprised of guides, tutorials, quickstarts, and reference docs. It's important to note that the [reference docs](https://supabase.com/docs/reference) are generated from spec files. As the DocSpec section above notes, we use documentation specs to generate human-readable docs.
```

Two quick examples to help clarify this:

> 1. If you want to update the doc for the supabase-js [https://supabase.com/docs/reference/javascript/auth-signup](<`signUp()`>) function, you would need to edit that in the supabase-js [https://github.com/supabase/supabase/edit/master/spec/supabase_js_v1_legacy.yml](spec file). The files generated from these spec files are included in the repo, so it can be confusing to know which file to edit. Just make sure you're not editing a file in a `_generated` directory, nor editing a file with a `custom_edit_url` field in the frontmatter at the top of the doc.

Reviewer 2: Might not need to mention the generated directory: https://github.com/supabase/supabase/pull/9078

Author: @dannykng the new instruction then should be something like:  "some files are generated, etc.....the source of truth for which file to edit is in the "Edit this page" link at the bottom?

Reviewer 2: Yep!
